### PR TITLE
Ensures renewal settings are reported correctly

### LINKED ISSUE #461: Check that client-visible renewal/deployment-enabled settings reflect reality
The reporter code that tells the user about what the client has done is being updated to indicate whether or not certs are going to be autorenewed (and autodeployed).  The way that it knows is by looking at the `ConfigObj` inside of the `RenewableCert` that was created when the cert was first obtained and enrolled.

I *think* this ConfigObj's notion of whether autorenewal and autodeployment are enabled is the same as the decision that `renewer.py` will come to when it runs later.  But we need to double-check that this is correct so that the user doesn't end up getting misleading information.

Cc @bradmw.

### PR DESCRIPTION
**Pull request by bmw**
Fixes #461. I opened the related #875, but this makes it so that only code in `storage.py` makes any decision on whether or not automatic renewal and deployment are enabled.

### CODE REVIEW

File: letsencrypt/storage.py [hunk 1]
```diff
@@ -439,6 +439,18 @@ def names(self, version=None):
         with open(target) as f:
             return crypto_util.get_sans_from_cert(f.read())
 
+    def autodeployment_is_enabled(self):
```
Comment (kuba) [inline letsencrypt/storage.py]: :+1: for factoring this out

File: letsencrypt/storage.py [hunk 2]
```diff
@@ -488,6 +499,18 @@ def ocsp_revoked(self, version=None):
         # certificate is not revoked).
         return False
 
+    def autorenewal_is_enabled(self):
```
Comment (kuba) [inline letsencrypt/storage.py]: ditto, :+1: for factoring this out

File: letsencrypt/storage.py (hunk 1 above)
Comment (kuba) [inline letsencrypt/storage.py]: This is missing unit test, and it seems to me that is not covered at all.

File: letsencrypt/storage.py (hunk 2 above)
Comment (kuba) [inline letsencrypt/storage.py]: This is missing unit test, and it seems to me that is not covered at all.

**Comment by kuba** [comment]
Seems to me to be a pure (but very good, :+1: !) refactor of the code. I guess that "fixes #461" comes from your better understanding of the code. Unless we can actually add some tests (maybe in the integration suite?) to detect #461 problems?

Please add those small tests as hinted above and it should be clear to merge :)

File: letsencrypt/storage.py (hunk 1 above)
Comment (bmw) [inline letsencrypt/storage.py]: Coverage was coming from `test_should_autodeploy`, but it doesn't hurt to have explicit unit tests.

File: letsencrypt/storage.py (hunk 2 above)
Comment (bmw) [inline letsencrypt/storage.py]: Coverage was coming from `test_should_autorenew`, but it doesn't hurt to have explicit unit tests.

**Comment by bmw** [comment]
This is a pure refactor, but I feel that this fixes #461 by removing the duplicated code that makes the decision of whether automatic renewal/deployment is enabled. If both `client.py` and `renewer.py` use the same code, they should reach the same decision.

**Comment by kuba** [comment]
LGTM. Thanks for explanation!

**Pull request merged**